spring-kafka/src/test/java/org/springframework/kafka/core/DefaultTransactionIdSuffixStrategyTests.java

```diff
@@ -0,0 +1,113 @@
+
+import java.util.Map;
+import java.util.Queue;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.kafka.test.utils.KafkaTestUtils;
+
+/**
+ * @author Ilya Starchenko
+ *
+ * @since 3.1.1
+ */
+class DefaultTransactionIdSuffixStrategyTests {
+
+	private DefaultTransactionIdSuffixStrategy suffixStrategy;
+
+	@BeforeEach
+	void setUp() {
+		suffixStrategy = new DefaultTransactionIdSuffixStrategy();
+	}
+
+	@Test
+	void testRetrieveSuffixWithoutCache() {
+		String txIdPrefix = "txIdPrefix";
+		String suffix = suffixStrategy.acquireSuffix(txIdPrefix);
+		assertThat(suffix).isNotNull();
+	}
+
+	@Test
+	void testRetrieveSuffixWithCache() {
+		String txIdPrefix = "txIdPrefix";
+		String suffix = "suffix";
+		assertThatNoException().isThrownBy(() -> suffixStrategy.releaseSuffix(txIdPrefix, suffix));
+	}
+
+
+	@Test
+	void testRetrieveSuffixWithCacheExhausted() {
+		String txIdPrefix = "txIdPrefix";
+		suffixStrategy.setMaxCache(2);
+		String suffix1 = suffixStrategy.acquireSuffix(txIdPrefix);
+		String suffix2 = suffixStrategy.acquireSuffix(txIdPrefix);
+		assertThatExceptionOfType(NoProducerAvailableException.class).isThrownBy(() -> suffixStrategy.acquireSuffix(txIdPrefix));
+	}
+
+	@Test
+	void testReturnSuffixWithCache() {
+		String txIdPrefix = "txIdPrefix";
+		suffixStrategy.setMaxCache(2);
+		String suffix = suffixStrategy.acquireSuffix(txIdPrefix);
+		assertThatNoException().isThrownBy(() -> suffixStrategy.releaseSuffix(txIdPrefix, suffix));
+	}
+
+	@SuppressWarnings("rawtypes")
+	@Test
+	void testReturnAllSuffixesWithCache() {
+		String txIdPrefix = "txIdPrefix";
+		suffixStrategy.setMaxCache(2);
+		String suffix1 = suffixStrategy.acquireSuffix(txIdPrefix);
+		String suffix2 = suffixStrategy.acquireSuffix(txIdPrefix);
+		assertThatNoException().isThrownBy(() -> suffixStrategy.releaseSuffix(txIdPrefix, suffix1));
+		Map<?, ?> suffixCache = KafkaTestUtils.getPropertyValue(suffixStrategy, "suffixCache", Map.class);
+		assertThat(suffixCache).hasSize(1);
+		Queue queue = (Queue) suffixCache.get(txIdPrefix);
+		assertThat(queue).hasSize(1);
+		assertThat(queue.poll()).isEqualTo(suffix1);
+	}
+
+	@Test
+	void testReset() {
+		String txIdPrefix = "txIdPrefix";
+		suffixStrategy.setMaxCache(2);
+		String suffix1 = suffixStrategy.acquireSuffix(txIdPrefix);
+		String suffix2 = suffixStrategy.acquireSuffix(txIdPrefix);
+		assertThatNoException().isThrownBy(() -> suffixStrategy.releaseSuffix(txIdPrefix, suffix1));
+		suffixStrategy.reset();
+		Map<?, ?> suffixCache = KafkaTestUtils.getPropertyValue(suffixStrategy, "suffixCache", Map.class);
+		assertThat(suffixCache).hasSize(0);
+	}
+
+	@Test
+	void testSetMaxCacheIsNegative() {
+		assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() -> suffixStrategy.setMaxCache(-1));
+	}
+
+	@Test
+	void testSetMaxCacheIsZero() {
+		assertThatNoException().isThrownBy(() -> suffixStrategy.setMaxCache(0));
+	}
+}
```

Comment:
It would be nice to make the test names consistent with the API name changes - `acquire` and `release.`. Also, since this is a new test and we are in JUnit 5, there is no need to prefix the test methods with the `test` literal. Cleaning those up in this new test class is good.

Reply:
Fixed.